msteams-platform/concepts/authentication/nested-authentication.md

```diff
@@ -0,0 +1,343 @@
+| **Errors** | Tom faces an sign-in error with Contoso due to an issue retrieving account information. Tom encounters a retry button that prompts for reauthentication. However, they discover that the system administrator has restricted access to Contoso. |
+
+## Configure nested authentication
+
+> [!NOTE]
+>
+> * As NAA is in developer preview and isn't supported by all host environments, ensure that you check the support status using the [isNAAChannelRecommended()](/javascript/api/@microsoft/teams-js/nestedappauth?) function and provide a fallback experience for unsupported environments.
+> * If the API returns the value as `true`, then call Microsoft Authentication Library (MSAL) for the NAA flow. If it returns `false`, continue to use your existing token retrieval method.
+
+To use nested authentication, follow these steps:
+
+1. [Register your SPA](#register-your-spa)
+1. [Add trusted brokers](#add-trusted-brokers)
+1. [Initialize public client app](#initialize-public-client-app)
+1. [Acquire your first token](#acquire-your-first-token)
+1. [Call an API](#call-an-api)
+
+### Register your SPA
+
+You must create a Microsoft Entra ID app registration for your add-in on Azure portal. The app registration must have a name, supported account type, and SPA redirect. Following the registration of your app, Azure portal generates a Microsoft Entra app registration ID.
+If your add-in requires additional app registration beyond NAA and SSO, see [register your single-page application.](/entra/identity-platform/scenario-spa-app-registration).
+
+### Add trusted brokers
+
+To enable nested authentication, your app must actively configure a redirect URI for your app. The redirect URI indicates to the Microsoft identity platform that your app can be brokered by supported hosts. The redirect URI of the app must be of type **Single-page application** and conform to the following scheme:
+
+```
+brk-<broker_application_id>://<your_domain>
+```
+
+Where,
+
+* <broker_application_id> is the alias of the broker or brokers you want to trust
+* <your_domain> is the fully qualified domain name where your app is hosted. For example, **brk-multihub://contoso.com**.
+
+If your app has been upgraded to run in Outlook and Microsoft365.com (in addition to Teams), then you need to only add one redirect URI:
+
+```http
+brk-multihub://<your_domain>
+```
+
+Your domain must include only the origin and not its subpaths. For example:
+
+✔️ brk-multihub://myapp.teams.microsoft.com <br>
+❌ brk-multihub://myapp.teams.microsoft.com/go
+
+### Initialize public client app
+
+> [!NOTE]
+> To ensure successful authentication, initialize TeamsJS before you initialize MSAL.
+
+Initialize MSAL and get an instance of the public client app to get access tokens, when needed.
+
+# [JavaScript](#tab/js1)
+
+```javascript
+import {
+  AccountInfo,
+  IPublicClientApplication,
+  createNestablePublicClientApplication,
```

Comment:
There should be one more parameter here:  supportsNestedAppAuth: boolean

Code reference in MSAL for that is here: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/a7d6315a17daf27bb0887fc03a3039a7ba6f57eb/samples/msal-browser-samples/OfficeAddin/src/taskpane/taskpane.js#L38

Reply:
Let's not provide the code reference since it's not maintained as a sample code. Instead, add it to the code sample already included in the document.

Reply:
@chrisyanasak @baljesingh As per the code reference shared, I've updated the code snippet for Initialize public client app. Please review and confirm.

Code reference:
<img width="640" alt="image" src="https://github.com/user-attachments/assets/cc95d8fa-8cba-4141-a0b7-a77d7d123100">

Added in the code snippet in the PR:
<img width="476" alt="image" src="https://github.com/user-attachments/assets/88a4a3bd-095f-4dbb-be8c-9d57593e5145">